src/interface/read_config_file.py

```diff
@@ -0,0 +1,108 @@
+import xml.etree.ElementTree as etree
+class ReadConfigFile():
+    
+    def __init__(self, fileconfig): 
+            
+            
+            self.xmld = etree.parse(fileconfig)             
+            self.root = self.xmld.getroot()
+
+        
+
+    def get_output_file(self):
+        """ 
+        Returns the outputfile path, name and extension values from config file
+        listoutput: variable to assign all values read from config file
+        """
+        list_output = []
+        for output in self.root.findall('Output'):
+            path = output.find('Path').text
+            name = output.find('Name').text         
+            extension = output.find('Extension').text   
+
+
+        list_output.append(path)
+        list_output.append(name)
+        list_output.append(extension)
+
+        
+        return list_output
+
+    def list_of_file_types_supported(self):
+        """     
+        Returns a list with all files types values supported, it will be also get from config file
+        listtypes: variable of list type, it will have assigned all supported files, this info is read from config file
+        """
+        list_types = []
+        
+        for solve in self.root.findall('./Output/Extension/type'):          
+            name_alg = solve.find('type').text
+            #print namealg
+            #listypes.append(typevalue)
+        
+
+    def get_list_of_algorithms(self):
+        """
+        Returns a list with all availables algorithms from config file, and the first item will always return
+        the active algorithm set for the game.
+        listalgorithms: variable of list type, it will have assigned all algorithms read from config file
+        """
+        list_algorithms = []
+        
+        for solve in self.root.findall('./Solve/Algorithm'):                        
+            if solve.find('Status').text == 'Active':
+                name_alg_active = solve.get('name')               
+            else:
+                name_alg = solve.get('name')
+                list_algorithms.append(name_alg)
+
+        list_algorithms.append(name_alg_active)
+        list_algorithms.reverse()
+        
+        return list_algorithms
+    
+    def get_list_of_generation_levels_names(self):
+
```

Comment:
this and the previous method looks identical with the only difference path used to iterate between the values, can we just have one method that abstract this functionality and avoid the duplication of code